contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -412,6 +444,35 @@ contract SetConfig is SetUp {
     );
   }
 
+  function testSetConfigRevertDueToInvalidBillingToken() public {
+    address[] memory billingTokens = new address[](1);
+    billingTokens[0] = address(linkToken);
+
+    AutomationRegistryBase2_3.BillingConfig[] memory billingConfigs = new AutomationRegistryBase2_3.BillingConfig[](1);
+    billingConfigs[0] = AutomationRegistryBase2_3.BillingConfig({
+      gasFeePPB: 5_000,
+      flatFeeMicroLink: 20_000,
+      priceFeed: 0x2222222222222222222222222222222222222222,
+      fallbackPrice: 2_000_000_000, // $20
+      minSpend: 100_000
+    });
+
+    bytes memory onchainConfigBytesWithBilling = abi.encode(cfg, billingTokens, billingConfigs);
+    bytes memory offchainConfigBytes = abi.encode(1234, ZERO_ADDRESS);
+    // deploy registry with OFF_CHAIN payout mode
+    registry = deployRegistry(AutoBase.PayoutMode.OFF_CHAIN);
+
+    vm.expectRevert(abi.encodeWithSelector(IAutomationRegistryMaster2_3.InvalidBillingToken.selector));
+    registry.setConfig(
```

Comment:
could we use `setConfigTypeSafe()` instead? It makes it easier to find bugs in future versions when we change how the `onChainConfigBytes` are decoded

Reply:
done